### ISSUE TITLE
git_config segfault problems and git_config_open_global proposal.

### ISSUE THREAD

**Issue by chobie** [libgit2/libgit2#210]
Hey, i have fun about git_config feature :)

well, i found some git_config plobrems in libgit2@4191d529245084fa92e869aadda72dd740403d3a, could you fix that? 
### 1. `git_config_open_file` segfault when specified non exist file path.

just as the title.

gcc -o case1 -lgit2 case1.c -g -O0
    #include <git2.h>

```
int main(int argc, char *argv[])
{
        git_config *config;
        git_config_open_file(&config,argv[1]);

        return 0;
}
```

running gdb.

```
# gdb ./case1 non_exist_file
.....
Breakpoint 2, backend_free (_backend=0x1001000c0) at ../../src/config_file.c:279
279             free(backend);
(gdb) p backend
$1 = (diskfile_backend *) 0x1001000c0
(gdb) p (diskfile_backend)backend
$2 = {
  parent = {
    cfg = 0x1001000c0, 
    open = 0x7fff5fbfee90, 
    get = 0x1000054b4 <git_config_free+62>, 
    set = 0x7fff5fbfee90, 
    foreach = 0x100100080, 
    free = 0xffffffff001000c0
  }, 
  var_list = {
    head = 0x100100140, 
    tail = 0x1001000c0
  }, 
  reader = {
    buffer = {
      data = 0x100140, 
      len = 140734799802080
    }, 
    read_ptr = 0x100005366 "�}�H�5��\002", 
    line_number = 0, 
    eof = 0
  }, 
  file_path = 0x0
}
(gdb) call (void)free(backend)
case1(62675) malloc: *** error for object 0x1001000c0: pointer being freed was not allocated
```
### 2. `git_config_open_file` segfault when reading a illegal config file.

i found segmentation fault in these two cases.

 case 2-1. unexpected blank line

```
[config]
    key = value

[configb]
    a = b
```

`git_config_open_file` crushed when founding a blank line.
blank lines will be ignored. 

case 2-2. illegal config file format

```
[level1 "level2" "unexpected level3"]
    unexpectedlevelfour = false
```

in this case,  `git_config_open_file` should stopping parse file and returns  error.
### 3. probably `git_config_open_global` should check `/etc/gitconfig` instead of `$HOME/gitconfig`

basically, git reads the config file form three location.

```
/etc/gitconfig
$HOME/.gitconfig
repository_dir/.git/config
```

so i think `git_config_open_global` should check `/etc/gitconfig`.
(sorry i don't know what is the better behavior on windows environment)

Regards,
Shuhei
